scripts/build.js

```diff
@@ -89,13 +93,95 @@ function checkChecksum(url, local_file) {
     })
 }
 
+/**
+ * Downloads an NPM package for use dynamically - so it will only be loaded and used for this single script.
+ * What it does under the hood is check for the npm package in the node modules, then in the npm list, and if
+ * it does not find it in either location, it will download the package at that point, adding it as a dev-dependency.
+ *
+ * It it downloads it dynamically, then it will return true. This is so you can delete the package once you are done,
+ * so it doesn't leave a zombie package in your node_modules. To remove the package, call npmDeleteRuntimePackage
+ *
+ * @param {*} package_name The name of the package you want to download (example: 'cmake-js')
+ * @param {*} package_version The version of the package to download - leave blank for latest. (example: '6.3.2')
+ * @returns True if the package was downloaded dynamically, otherwise false.
+ */
+function npmDownloadAndInstallRuntimePackage(package_name, package_version=null) {
+    console.log("Looking for " + package_name + " as a dependency...");
+
+    // Do we have it in node modules? If so, then use that!
+    try {
+        if (fs.existsSync("./node_modules/" + package_name)) {
```

Comment:
you can use `require.resolve` if you want to check for a module and aren't specifically looking in the immediate node_modules directory.